lib/puppet/network/http/rack/rest.rb

```diff
@@ -103,7 +103,21 @@ def client_cert(request)
     # NOTE: The SSL_CLIENT_CERT environment variable will be the empty string
     # when Puppet agent nodes have not yet obtained a signed certificate.
     if cert.nil? || cert.empty?
-      nil
+      # When running with unicorn, the SSL_CLIENT_CERT variable is not available
+      # in the environment, therefore we have to pass a header: 'X-SSL-Client-Cert'
+      cert = request.env['HTTP_X_SSL_CLIENT_CERT']
```

Comment:
I tried with request.params, but the extensions were an empty hash, reverting back to request.env, and the hash was filled again.